yb-thirdparty-common.sh

```diff
@@ -75,7 +75,6 @@ ensure_correct_mac_architecture() {
   fi
   if [[ $YB_TARGET_ARCH != "x86_64" && $YB_TARGET_ARCH != "arm64" ]]; then
     fatal "Invalid value of YB_TARGET_ARCH on macOS (expected x86_64 or arm64): $YB_TARGET_ARCH"
-    exit 1
```

Comment:
Why was this removed?

Reply:
From the commit message of @vrajat `Remove exit 1 as fatal already exits. Caught by shellcheck`, it seems like `exit 1` won't be called. Should this be reverted?

Reply:
Oh, I see. That is a call to fatal, not just log message. I just wanted to make sure it was intentional.